x-pack/plugin/otel-data/src/main/resources/component-templates/semconv-resource-to-ecs@mappings.yaml

```diff
@@ -170,6 +170,10 @@ template:
       kubernetes.node.hostname:
         type: alias
         path: resource.attributes.k8s.node.hostname
+      # Below is needed for the k8s events filtering per namespace
+      resource.attributes.k8s.namespace.name:
```

Comment:
I don't think that will work because the field is already defined in line 74 as a keyword field. Defining the same field multiple times with different field types isn't possible in ES.

Could we instead copy `body.object.regarding.namespace` to `resource.attributes.k8s.namespace.name` in the OTel collector?

Reply:
Yes we can do that. I will close this PR and try the change in collector side

Reply:
@felixbarny I rushed to say that can be done , by testing proved me wrong. 

The k8sobject receiver is used for the events see https://github.com/elastic/opentelemetry/blob/main/resources/kubernetes/operator/helm/values.yaml#L158-L164

And used the attributes processor like:

```bash
        resource/k8sevents:
          attributes:
            - key: k8s.namespace.name
              from_attribute: body_structured.object.regarding.namespace
              action: insert
```

But I dont think is a valid scenario one processor to interact with other, or to rephrase it does not seem that the attribute processor can read the event logs. 

I will create a seperate issue to enhance events with namespace and will research there for any solution